utils/src/main/java/software/amazon/awssdk/utils/LambdaSystemSetting.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.utils;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * The system properties provided by aws lambda environment.
+ */
+@SdkProtectedApi
+public enum LambdaSystemSetting implements SystemSetting {
```

Comment:
Do we actually need this to be here? We should try to avoid adding unnecessary public/protected APIs because more public API surface area means more maintenance and less flexibility to change it